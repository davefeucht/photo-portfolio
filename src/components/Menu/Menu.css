.menu {
  grid-area: content;
  height: 100%;
  width: 300px;
  background-color: black;
  transition: opacity 0.5s ease-in-out;
  z-index: 10;

  color: #FFFFFF;
  font-size: 22px;
}

.menu.open {
  opacity: 0.9;
}
.menu.closed {
  opacity: 0;
  pointer-events: none;
}

.menu ul {
    list-style: none;
}