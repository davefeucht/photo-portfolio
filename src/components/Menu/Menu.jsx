/** ***************
* Menu component implements the menu of the application
**************** */

import './Menu.css';

import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React from 'react';

import MenuLink from '../MenuLink/MenuLink.jsx';

const Menu = ({ stateStore }) => {
    const {
        menuState, toggleMenuState, categoryList, pages
    } = stateStore;
    return (
        <div className={`menu ${menuState}`}>
            <ul>
                <li onClick={() => toggleMenuState()}>
                    <MenuLink href="" text="Home" />
                </li>
                <li>
                    Categories
                    <ul>
                        {categoryList.map(category => {
                            return (
                                <li key={`li_${category.name}_${category.id}`} onClick={() => toggleMenuState()}><MenuLink key={`${category.name}_${category.id}`} href={`category/${category.id}`} text={category.name} /></li>
                            );
                        })}
                    </ul>
                </li>
                <li>
                    Info
                    <ul>
                        {pages.map(page => {
                            return (
                                <li key={`li_${page.title.rendered}`} onClick={() => toggleMenuState()}><MenuLink key={`${page.title.rendered}_${page.id}`} href={`page/${page.id}`} text={page.title.rendered} /></li>
                            );
                        })}
                    </ul>
                </li>
            </ul>
        </div>
    );
};

Menu.displayName = 'Menu';

Menu.propTypes = {
    stateStore: PropTypes.object.isRequired
};

export default observer(Menu);
