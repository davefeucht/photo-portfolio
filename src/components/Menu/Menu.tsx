/** ***************
* Menu component implements the menu of the application
**************** */

import './Menu.css';

import { observer } from 'mobx-react';
import * as React from 'react';

import { Store } from '../../utils/types';
import MenuLink from '../MenuLink/MenuLink';

interface MenuProps {
    stateStore: Store
}

const Menu: React.FC<MenuProps> = ({ stateStore }) => {
    const {
        menuState, toggleMenuState, categoryList, pages
    } = stateStore;
    return (
        <div className={`menu ${menuState}`}>
            <ul>
                <li onClick={() => toggleMenuState()}>
                    <MenuLink href="" text="Home" />
                </li>
                <li>
                    Categories
                    <ul>
                        {categoryList.map(category => {
                            return (
                                <li key={`li_${category.name}_${category.id}`} onClick={() => toggleMenuState()}><MenuLink key={`${category.name}_${category.id}`} href={`category/${category.id}`} text={category.name} /></li>
                            );
                        })}
                    </ul>
                </li>
                <li>
                    Info
                    <ul>
                        {pages.map(page => {
                            return (
                                <li key={`li_${page.title.rendered}`} onClick={() => toggleMenuState()}><MenuLink key={`${page.title.rendered}_${page.id}`} href={`page/${page.id}`} text={page.title.rendered} /></li>
                            );
                        })}
                    </ul>
                </li>
            </ul>
        </div>
    );
};

Menu.displayName = 'Menu';

export default observer(Menu);
