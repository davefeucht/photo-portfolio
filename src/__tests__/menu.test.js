import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Menu from '../components/Menu/Menu';
import stateStore from '../StateStore/store';

test('CategoryThumbnail displays', async () => {
    const store = new stateStore();
    let component;
    let tree;
    component = renderer.create(
        <MemoryRouter>
            <Menu stateStore={store}></Menu>
        </MemoryRouter>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
