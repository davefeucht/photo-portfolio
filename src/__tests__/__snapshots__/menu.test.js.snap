// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryThumbnail displays 1`] = `
<div
  className="menu closed"
>
  <ul>
    <li
      onClick={[Function]}
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="menu-link"
        >
          Home
        </div>
      </a>
    </li>
    <li>
      Categories
      <ul />
    </li>
    <li>
      Info
      <ul />
    </li>
  </ul>
</div>
`;
